text/000-rustdoc-cfgs-handling.md

```diff
@@ -0,0 +1,281 @@
+Rustdoc: stabilization of the `doc(cfg*)` attributes
+
+- Features Name: `doc_cfg`
+- Start Date: 2022-12-07
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#43781](https://github.com/rust-lang/rust/issues/43781)
+
+
+# Summary
+[summary]: #summary
+
+This RFC aims at providing rustdoc users the possibility to add visual markers to the rendered documentation to know under which conditions an item is available (currently possible through the following unstable features: `doc_cfg`, `doc_auto_cfg` and `doc_cfg_hide`).
+
+It does not aim to allow having a same item with different `cfg`s to appear more than once in the generated documentation.
+
+It does not aim to document items which are *inactive* under the current configuration (i.e., “`cfg`'ed out”).
+
+# Motivation
+[motivation]: #motivation
+
+The goal of this RFC is to stabilize the possibility to add visual markers to the rendered documentation to know under which conditions an item is available.
+
+Providing this information to users will solve a common issue: “Why can I see this item in the documentation and yet can't use it in my code?”.
+The end goal being to provide this information automatically so that the documentation maintenance cost won't increase.
+
+
+# Guide-level explanation
```

Comment:
I mentioned this as a regular comment, but I'll move it to a review so it gets noticed: I think that the guide-level explanation should clarify the `cfg(any(doc, ...))` trick for making `auto_cfg` work properly, since it's not immediately clear otherwise. Basically, conditional compilation is still applied before rustdoc runs, meaning you have to ensure that rustdoc can see code to document it. Then, because `auto_cfg` by default ignores `cfg(doc)`, only the other `cfg`s are documented.

Reply:
It was not ignored. You can see the RFC mentioning this [here](https://github.com/rust-lang/rfcs/pull/3631/files#diff-2a397ac636038db2e30773d6b828f499f05e0b511b59da0522ac75e2a9868001R16). If it's not clear enough, how would you word it?